### PR TITLE
track changes/atomic commit

### PR DESCRIPTION
**Pull request by earonesty**
readers don't see changes writers make until the __exit__ func is called

### CODE REVIEW

File: omen2/object.py [hunk 1]
```diff
@@ -188,65 +269,95 @@ def _collect_cascade(self):
...
-            if len(db_row) > 1:
-                raise OmenMoreThanOneError("more than 1 %s in the db" % self.__class__)
-        self._update(db_row[0])
-
     def __enter__(self):
-        if self._meta and self._meta.table:
-            self._meta.lock.acquire()
-            self._meta.locked = True
+        if self.__meta and self.__meta.table:
+            self.__meta.lock.acquire()
```
Comment (higleyc) [inline omen2/object.py:334]: thought: perhaps as a safety, use a large timeout/throw to prevent hypothetical deadlocks

File: omen2/object.py [hunk 2]
```diff
@@ -188,65 +269,95 @@ def _collect_cascade(self):
...
+            self.__meta.locked = True
+            self.__meta.suppress_set_changes = False
```
Comment (higleyc) [inline omen2/object.py:336]: just confirming that this isn't necessary, i.e. `suppress_set_changes` should always already be `False`

Comment (earonesty) [reply omen2/object.py:336]: correct, removed in the chris-mr

File: omen2/object.py (hunk 1 above)
Comment (earonesty) [reply omen2/object.py:334]: added to the "from coderev" branch

**Pull request merged**